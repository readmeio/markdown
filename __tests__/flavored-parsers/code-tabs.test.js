import { mdast, hast } from '../../index';

describe('Code Tabs', () => {
  it('can parse code tabs', () => {
    const md = `
\`\`\`
First code block
\`\`\`
\`\`\`
Second code block
\`\`\`
`;

    expect(mdast(md)).toMatchSnapshot();
  });

  it('can parse lang and meta', () => {
    const md = `
\`\`\`javascript First Title
First code block
\`\`\`
\`\`\`text
Second code block
\`\`\`
`;
    const ast = mdast(md);

    expect(ast.children[0].children[0]).toStrictEqual(
      expect.objectContaining({ lang: 'javascript', meta: 'First Title' })
    );
    expect(ast.children[0].children[1]).toStrictEqual(expect.objectContaining({ lang: 'text', meta: '' }));
  });

  it('allows code tabs within html blocks', () => {
    const md = `
<p>

\`\`\`
First code block
\`\`\`
\`\`\`
Second code block
\`\`\`

</p>
`;

    expect(hast(md)).toMatchSnapshot();
  });
});
