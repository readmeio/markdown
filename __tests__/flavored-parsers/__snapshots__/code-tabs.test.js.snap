// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Tabs allows code tabs within html blocks 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "
",
        },
      ],
      "properties": Object {},
      "tagName": "p",
      "type": "element",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "First code block
",
                },
              ],
              "properties": Object {
                "meta": "",
              },
              "tagName": "code",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "pre",
          "type": "element",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Second code block
",
                },
              ],
              "properties": Object {
                "meta": "",
              },
              "tagName": "code",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "pre",
          "type": "element",
        },
      ],
      "properties": Object {
        "className": Array [
          "code-tabs",
        ],
      },
      "tagName": "div",
      "type": "element",
    },
    Object {
      "type": "text",
      "value": "
",
    },
    Object {
      "children": Array [],
      "properties": Object {},
      "tagName": "p",
      "type": "element",
    },
  ],
  "data": Object {
    "quirksMode": false,
  },
  "type": "root",
}
`;

exports[`Code Tabs can parse code tabs 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "className": "tab-panel",
          "data": Object {
            "hName": "code",
            "hProperties": Object {
              "lang": undefined,
              "meta": "",
            },
          },
          "lang": undefined,
          "meta": "",
          "type": "code",
          "value": "First code block",
        },
        Object {
          "className": "tab-panel",
          "data": Object {
            "hName": "code",
            "hProperties": Object {
              "lang": undefined,
              "meta": "",
            },
          },
          "lang": undefined,
          "meta": "",
          "type": "code",
          "value": "Second code block",
        },
      ],
      "className": "tabs",
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": Array [
            "code-tabs",
          ],
        },
      },
      "type": "code-tabs",
    },
  ],
  "type": "root",
}
`;
